### PR TITLE
[JENKINS-72222] Fix inconsistent wording between login page and security configuration

### PR DESCRIPTION
**Pull request by mustafau**
The option in the login page was reworded as "**Keep me signed in**" in the past however the wording in the security configuration was never updated. This PR only changes the help text and the Javadoc to refer to the correct option. 

I'm marking this PR as draft but I believe it's complete. I just want to make sure this is the best way to address the inconsistency. 

See [JENKINS-72222](https://issues.jenkins.io/browse/JENKINS-72222).

### Testing done

Screen Shot 2023-10-22 at 22 39 14
Screen Shot 2023-10-22 at 22 38 40

### Proposed changelog entries

- JENKINS-72222, refer to the correct option in the help text

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by daniel-beck** [comment]
Might be better to change the entire UI label instead of only the help? Perhaps phrased as

> Disable "Keep me signed in"

(with typographic quotes) to make it easier to read?

**Comment by mustafau** [comment]
@daniel-beck I've updated the label. Here's updated screenshot:

Screen Shot 2023-10-27 at 00 54 36

File: core/src/main/resources/hudson/security/GlobalSecurityConfiguration/index.properties
Comment (daniel-beck) [inline core/src/main/resources/hudson/security/GlobalSecurityConfiguration/index.properties:1]: This is what I meant when I wrote typographic quotes:

```suggestion
Disable\ remember\ me=Disable “Keep me signed in”
```

Comment (mustafau) [reply core/src/main/resources/hudson/security/GlobalSecurityConfiguration/index.properties:1]: Thanks. I didn't know the difference. 

Silly question: should I use typographic quotes in Javadoc? I probably shouldn't right?

Comment (daniel-beck) [reply core/src/main/resources/hudson/security/GlobalSecurityConfiguration/index.properties]: No, but it's preferable for regular UI text.

**Review by daniel-beck — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
